WordPress/openverse
# Fix image and audio result focus rings

## Fixes



Fixes #3525 by @fcoveram
Fixes #3193 by @fcoveram 
Related to #4911 by @zackkrida 

## Description


This PR updates `focus-bold-filled` to set the inner ring to the color of the page background. It also updates the focus ring for the non-square image result.

The inner ring looks narrower in dark theme than it does in the light theme, ~~so I updated the width for the dark theme.~~ but I think it just seems so, and the ring is actually of the same width.

Initially, I tried to use `focus-slim-offset` class, but the way it creates the offset made the rounded corners look bad: they were wider than the ring itself. That's why I updated it to use `focus-slim-tx`, like the audio track uses, but also added padding and negative margin to add "offset".

Edited to add: with the current implementation of `focus` classes based on `*` selector, it is not possible to use `focus-bold-filled` with `sm:` prefixes, as required by the designs (style for image result grid is `focus-slim` on `xs` breakpoint, but `fold-filled` on larger breakpoints). I changed the `*focus-bold-filled` to `.focus-bold-filled` to match the class exactly. This class should only be used with `focus-visible:` prefix, to prevent css interference.

I had to fix the z-index to stop audio focus ring overlapping the audio instructions snackbar.

## Testing Instructions

Check the result focus rings for image, audio and all content results.
You can use the storybook to view [image cell](https://docs.openverse.org/_preview/4959/storybook/?path=/story/components-vimagecell--default&args=aspectRatio:square&globals=viewport:sm;theme:dark) and the [audio result](https://docs.openverse.org/_preview/4959/storybook/?path=/story/components-audio-track--box-small&globals=viewport:xs;theme:dark)s

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
